### PR TITLE
Sheryl Stillman Final

### PR DESCRIPTION
**Pull request by stillsheryl**
Overall I feel that the final went pretty well. I went back and forth about where to put the action to destroy the relationship between the doctor and patient, and ended up with a controller so I could make a destroy action that made sense, rather than “destroying” in the patient controller, since the relationship is being destroyed and not the patient itself. This action is the messiest, and I think there's a better way to do it, but I'm not sure how.

I tried to adhere to MVC and REST as much as possible, and wonder if the `working_hospital` method is necessary, or if I can just put `@doctor.hospital.name` in the view.

I finished the extension and would have used a gem to check the order, but went to CSS and was able to successfully test the order. 

If I had more time I would go through all my tests and make sure all edge cases are tested. I tried to be aware of them as I went, but if time allowed I'd go through to double check all test coverage. SimpleCov says 100%, so that's nice!

EDIT: Also, just as I turned this in I realized I probably should have made a method in the DoctorPatient model to retrieve the specific one I needed.

### CODE REVIEW

File: app/controllers/doctor_patients_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class DoctorPatientsController < ApplicationController
+  def destroy
+    doc_relationship = DoctorPatient.select("doctor_id = #{params[:id]}").where("patient_id = #{params[:patient_id]}").pluck(:id).first
```
Comment (timomitchel) [inline app/controllers/doctor_patients_controller.rb:3]: This would be better off as `DoctorPatient.where("patient_id = #{params[:patient_id]}, doctor_id = #{params[:id]}").destroy_all` -- and in the case we know we only want to destroy one record you could also do a `.first.destroy` instead of `destroy_all`, but this is less efficient.

File: app/models/doctor.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Doctor < ApplicationRecord
+  belongs_to :hospital
+  has_many :doctor_patients
+  has_many :patients, through: :doctor_patients
+
+  def working_hospital
```
Comment (timomitchel) [inline app/models/doctor.rb:6]: You could call this in the view.

File: app/models/hospital.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Hospital < ApplicationRecord
+  has_many :doctors
+
+  def doctor_count
+    doctors.count
+  end
+
+  def unique_universities
+    self.doctors.select(:university).order(:university).pluck(:university).uniq
```
Comment (timomitchel) [inline app/models/hospital.rb:9]: Would have rather seen you use `.distinct` which is ActiveRecord as opposed to `.uniq` which is ruby.

**Comment by timomitchel** [comment]
Thank you for the thorough reflection. I left some comments!
